bazel/envoy_build_system.bzl

```diff
@@ -20,12 +20,45 @@ def envoy_copts(repository, test = False):
         repository + "//bazel:dbg_build": ["-ggdb3"],
     }) + select({
         repository + "//bazel:disable_tcmalloc": [],
+        "//bazel/osx:darwin": [],
         "//conditions:default": ["-DTCMALLOC"],
     }) + select({
         repository + "//bazel:disable_signal_trace": [],
         "//conditions:default": ["-DENVOY_HANDLE_SIGNALS"],
     })
 
+# Compute the final linkopts based on various options.
+def envoy_linkopts():
+    return select({
+        "//bazel/osx:darwin": [],
```

Comment:
OS X system libraries are distributed exclusively as dynamically loaded libraries. So, all the thirdparty dependencies (e.g. ci/build_container/build_recipes) are linked statically, but the binary links libc++ and libSystem dynamically. That second library dynamically links libraries for libpthread and libuuid.